service/src/main/java/skills/skillLoading/SkillsLoader.groovy

```diff
@@ -406,10 +410,25 @@ class SkillsLoader {
     }
 
     @Transactional(readOnly = true)
-    SkillSummary loadSkillSummary(String projectId, String userId, String crossProjectId, String skillId) {
+    SkillSummary loadSkillSummary(String projectId, String userId, String crossProjectId, String skillId, String subjectId) {
         ProjDef projDef = getProjDef(userId, crossProjectId ?: projectId)
         SkillDefWithExtra skillDef = getSkillDefWithExtra(userId, crossProjectId ?: projectId, skillId, [SkillDef.ContainerType.Skill, SkillDef.ContainerType.SkillsGroup])
 
+        String nextSkillId;
+        String prevSkillId;
+
+        if(subjectId) {
+            SkillSubjectSummary subject = loadSubject(projectId, userId, subjectId, publicProps.getInt(PublicProps.UiProp.maxSkillVersion), true)
```

Comment:
``loadSubject`` method is very expensive; suggesting to execute 2 sql statements to locate prev and next skillIds (1 for previous and 1 for next skillId)

Reply:
Changed to a new method that runs 2 SQL statements